### DISCUSSION TITLE
ProtoAdapter.STRING.encodeWithTag adding extra characters

### DISCUSSION THREAD

**Discussion by venkatduddu** [square/wire#2087] [Q&A]
Hi, 

I'm trying to encode the data using ProtoAdapter.STRING for value `618d2cde-31a6-485d-8231-36a322b4dbc8` and when I decode the same data I see data is prefixed with `\n$` i.e decoded data looks like `\n$618d2cde-31a6-485d-8231-36a322b4dbc8`. Is this expected?

Actual call looks like `ProtoAdatapter.STRING.encodeWithTag(protoWriter, tagValue, "618d2cde-31a6-485d-8231-36a322b4dbc8");`

When I debugged this further I found that in `internal inline fun <E> ProtoAdapter<E>.commonEncodeWithTag`after `writer.writeTag(tag, fieldEncoding)` `\n` is being added and writer.writeVarint32(encodedSize(value)) adds `$`

Can I get some help in avoiding those chars?

**Comment by JakeWharton** [comment]
The protobuf spec mandates all values have a tag and all non-scalar's be length prefixed. So no you cannot avoid those characters. What are you trying to do?

**Reply by venkatduddu** [reply]
Hi @JakeWharton 

This is what I'm trying to do:

1. I receive encoded data along with schema associated. Parse the data using the ProtoParser, ProtoReader and populate data into Map<String, Object>
2. Using the above Map write back encoded data using a different schema (in this case most of the fields are common from the original schema [tags might differ from original schema] and some additional fields something like enriched data).
3. Test the encoded data (by parsing similar to step#1) from step#2 to see data is copied properly and additional fields are populated or not.

In step#3 I'm seeing those extra characters.

Please note that I don't have the flexibility of using generated classes to encode, decode protobuf.

**Reply by venkatduddu** [reply]
To add more context to the problem, this only happens when we have schema referencing the message from another proto file, for ex:

```
syntax = "proto3";

import "protos/imports/webrequest.proto";

message SearchRequestEnriched {
  string client_ip = 1;
  string user_agent = 2;
  string query = 101;
  int32 page_number = 102;
  int32 result_per_page = 103;
  optional WebRequest webReq = 104;
}

syntax = "proto3";

message WebRequest {
  string requestUuid = 1;
}
```

So while I'm encoding data for SearchRequestEnriched first I encode the WebRequest and write the encoded bytes like this.

ProtoAdapter.BYTES_VALUE.encodeWithTag(enrichedSearchRequestProtoWriter, tagValue, byteStringFromWebRequestEncoding);
